File: 1072-flip-columns-for-maximum-number-of-equal-rows/1072-flip-columns-for-maximum-number-of-equal-rows.cpp
class Solution {
public:
    int maxEqualRowsAfterFlips(vector<vector<int>>& matrix) {
    int n = matrix.size();
    int m = matrix[0].size();
    map<string,int> mp;
    for(int i=0; i<n; i++) {
        string originalMat="";
        string complimentMat="";
        for(int j=0; j<m; j++) {
            originalMat.push_back(matrix[i][j] + '0');
            complimentMat.push_back((1 ^ matrix[i][j]) + '0');
        }
        mp[originalMat]++;
        mp[complimentMat]++;
    }
    int ans = 0;
    for(auto x : mp) {
        ans = max(ans, x.second);
    }
    
     return ans;
    }
};

File: 1072-flip-columns-for-maximum-number-of-equal-rows/1072-flip-columns-for-maximum-number-of-equal-rows.cs
public class Solution {
    public int MaxEqualRowsAfterFlips(int[][] matrix) {
       
        int n = matrix.Length;
        int m = matrix[0].Length;
        Dictionary<string,int> mp = new Dictionary<string,int>();
        for(int i=0; i<n; i++) {
            string originalMat="";
            string complimentMat="";
            for(int j=0; j<m; j++) {
                originalMat += matrix[i][j].ToString();
                complimentMat += (1 ^ matrix[i][j]).ToString();
            }
            if (mp.ContainsKey(originalMat)) {
                mp[originalMat]++;
            } else {
                mp[originalMat] = 1;
            }
            if (mp.ContainsKey(complimentMat)) {
                mp[complimentMat]++;
            } else {
                mp[complimentMat] = 1;
            }
        }
        int ans = 0;
        foreach(var x in mp) {
            ans = Math.Max(ans, x.Value);
        }
        return ans;
    }
}
    
